#!/usr/env/python
PACKAGE = "my_dyn_rec"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#List of parameters 
gen.add("position_x", double_t, 0, "center position x", -5.0, -20.0, 22.0) # default min max
gen.add("position_y", double_t, 0, "center position y",  0.0, -20.0, 20.0)
gen.add("cell_size", double_t, 0, "size of a grid cell", 0.5, 0.01, 5.0)
gen.add("length_x", double_t, 0, "In meters",  80.0, 0.1, 200.0)
gen.add("length_y", double_t, 0, "In meters",  80.0, 0.1, 200.0)
gen.add("intensity_factor", double_t, 0, "Multiply intensity",  0.2, 0.001, 10.0)
gen.add("height_factor", double_t, 0, "Multiply height",  1.0, 0.001, 10.0)
gen.add("cloud_in_topic", str_t, 0, "A LIDAR topic to subscribe. Restart needed if it changes.", "/demo/nonground")
# gen.add("cloud_in_topic", str_t, 0, "A LIDAR topic to subscribe. Restart needed if it changes.", "/left_os1/os1_cloud_node/points")
gen.add("mapi_topic_name", str_t, 0, "Intensity map topic name. Restart needed if it changes.",  "/lidargrid_i")
gen.add("maph_topic_name", str_t, 0, "Height map topic name. Restart needed if it changes.",  "/lidargrid_h")
gen.add("debug", bool_t, 0, "publish debug topics too", True)

exit(gen.generate(PACKAGE, "my_dyn_rec", "MyParams"))